src/views/preview/preview.jsx

```diff
@@ -10,35 +11,125 @@ const sessionActions = require('../../redux/session.js');
         if (pathname[pathname.length - 1] === '/') {
             pathname = pathname.substring(0, pathname.length - 1);
         }
         const path = pathname.split('/');
         const projectId = path[path.length - 1];
-        this.props.getProjectInfo(projectId);
-        this.props.getRemixes(projectId);
-        // get comments
-        // get studios
-
+        if (this.props.sessionStatus !== prevProps.sessionStatus &&
+            this.props.sessionStatus === sessionActions.Status.FETCHED) {
+            if (this.props.user) {
+                const username = this.props.user.username;
+                const token = this.props.user.token;
+                this.props.getProjectInfo(projectId, token);
+                this.props.getRemixes(projectId, token);
+                this.props.getStudios(projectId, token);
+                this.props.getFavedStatus(projectId, username, token);
+                this.props.getLovedStatus(projectId, username, token);
+            } else {
+                this.props.getProjectInfo(projectId);
+                this.props.getRemixes(projectId);
+                this.props.getStudios(projectId);
+            }
+            
+        }
+        if (this.props.projectInfo.id !== prevProps.projectInfo.id) {
+            this.initCounts(this.props.projectInfo.stats.favorites, this.props.projectInfo.stats.loves);
+            this.handlePermissions();
+            if (this.props.projectInfo.remix.root !== null) {
+                this.props.getCreditInfo(this.props.projectInfo.remix.root);
+            }
+        }
     }
-    componentDidUpdate (prevProps) {
-        if (this.props.projectInfo.id !== prevProps.projectInfo.id && this.props.projectInfo.remix.root !== null) {
-            this.props.getCreditInfo(this.props.projectInfo.remix.root);
+    handleFavoriteToggle () {
+        this.props.setFavedStatus(
+            !this.props.faved,
+            this.props.projectInfo.id,
+            this.props.user.username,
+            this.props.user.token
+        );
+        if (this.props.faved) {
+            this.setState(state => ({
+                favoriteCount: state.favoriteCount - 1
+            }));
+        } else {
+            this.setState(state => ({
+                favoriteCount: state.favoriteCount + 1
+            }));
         }
-
+    }
+    handleLoveToggle () {
+        this.props.setLovedStatus(
+            !this.props.loved,
+            this.props.projectInfo.id,
+            this.props.user.username,
+            this.props.user.token
+        );
+        if (this.props.loved) {
+            this.setState(state => ({
+                loveCount: state.loveCount - 1
+            }));
+        } else {
+            this.setState(state => ({
+                loveCount: state.loveCount + 1
+            }));
+        }
+    }
+    handlePermissions () {
+        // TODO: handle admins and mods 
+        if (this.props.projectInfo.author.username === this.props.user.username) {
```

Comment:
Another one that might be good to consolidate either into a single prop or into a method call like `isAuthor`.

Reply:
Agreed. I've left them as-is for now as I think there will be more statuses once we get into unshared and admin/mods.